src/main/java/org/jabref/logic/ai/embeddings/MVStoreEmbeddingStore.java

```diff
@@ -0,0 +1,197 @@
+package org.jabref.logic.ai.embeddings;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.PriorityQueue;
+import java.util.UUID;
+import java.util.function.Predicate;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import org.jabref.gui.DialogService;
+import org.jabref.logic.ai.FileEmbeddingsManager;
+
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.store.embedding.CosineSimilarity;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import dev.langchain4j.store.embedding.EmbeddingSearchRequest;
+import dev.langchain4j.store.embedding.EmbeddingSearchResult;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import dev.langchain4j.store.embedding.RelevanceScore;
+import dev.langchain4j.store.embedding.filter.Filter;
+import dev.langchain4j.store.embedding.filter.comparison.IsEqualTo;
+import dev.langchain4j.store.embedding.filter.comparison.IsIn;
+import jakarta.annotation.Nullable;
+import org.h2.mvstore.MVMap;
+import org.h2.mvstore.MVStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.util.Comparator.comparingDouble;
+
+/**
+ * A custom implementation of langchain4j's {@link EmbeddingStore} that uses a {@link MVStore} as an embedded database.
+ * <p>
+ * Every embedding has 3 fields: float array (the embedding itself), file where it was generated from, and the embedded
+ * string (the content). Each of those fields is stored in a separate {@link MVMap}.
```

Comment:
Why? I think, a `record` with three fields can be used.

Reply:
Through a lot investigation I found out that `MVStore` doesn't support mutable values.

And there seems no method to "forcefully" update the storage.

I've written a question on their GitHub to check (https://github.com/h2database/h2database/issues/4090), but I don't think they have active community.

Propose to leave it as is, + I'll write the comment why we did that, + update ADR (https://github.com/JabRef/jabref/blob/6e2d60c14273b22f5ae1e90ac5dfb760954ef5cb/docs/decisions/0033-store-chats-in-mvstore.md)

Reply:
> Through a lot investigation I found out that MVStore doesn't support mutable values.

Is there any code that I can read?